mllib/src/main/scala/org/apache/spark/mllib/regression/GeneralizedLinearAlgorithm.scala

```diff
@@ -287,7 +282,7 @@ abstract class GeneralizedLinearAlgorithm[M <: GeneralizedLinearModel]
         if (useFeatureScaling) {
           input.map(lp => (lp.label, scaler.transform(lp.features))).cache()
         } else {
-          input.map(lp => (lp.label, lp.features))
+          input.map(lp => (lp.label, lp.features)).cache()
```

Comment:
I think this omission was intentional: If input is cached, then there's no real need to cache this tiny transform.  If you keep as is, then you can just check at the end to see if "data" is cached before trying to unpersist it.